lib/matplotlib/texmanager.py

```diff
@@ -451,30 +429,25 @@ def make_dvi_preview(self, tex, fontsize):
         if (DEBUG or not os.path.exists(dvifile) or
                 not os.path.exists(baselinefile)):
             texfile = self.make_tex_preview(tex, fontsize)
-            outfile = basefile + '.output'
-            command = self._get_shell_cmd(
-                'cd "%s"' % self.texcache,
-                'latex -interaction=nonstopmode %s > "%s"' %
-                (os.path.split(texfile)[-1], outfile))
+            command = [str("latex"), "-interaction=nonstopmode",
+                       os.path.basename(texfile)]
             mpl.verbose.report(command, 'debug')
-            exit_status = os.system(command)
             try:
-                with open(outfile) as fh:
-                    report = fh.read()
-
-            except IOError:
-                report = 'No latex error report available.'
-            if exit_status:
+                report = subprocess.check_output(command,
+                                                 cwd=self.texcache,
+                                                 stderr=subprocess.STDOUT)
+            except subprocess.CalledProcessError as exc:
                 raise RuntimeError(
                     ('LaTeX was not able to process the following '
-                     'string:\n%s\nHere is the full report generated by '
-                     'LaTeX: \n\n' % repr(tex)) + report)
-            else:
-                mpl.verbose.report(report, 'debug')
+                     'string:\n%s\n\n'
+                     'Here is the full report generated by LaTeX:\n%s '
+                     '\n\n' % (repr(tex.encode('unicode_escape')),
```

Comment:
Good question. This was like this in one of the calls the current master and I took that over in all new `subprocess.check_output`s